setup/src/Magento/Setup/Module/Di/Code/Reader/InvalidFileException.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Magento\Setup\Module\Di\Code\Reader;
+
+class InvalidFileException extends \InvalidArgumentException
```

Comment:
For `Magento\Setup\Module\Di\Code\Reader\FileClassScanner::__construct` I'd suggest to go with `\Magento\Framework\Exception\FileSystemException`. It's used the same way in `\Magento\Framework\Filesystem\Directory\Write::assertIsFile`.

For `Magento\Setup\Module\Di\Code\Reader\FileClassScanner::isBracedNamespace`, I'm a bit undecided which of the existing exeptions is the best fit. Maybe `\Magento\Framework\Exception\NotFoundException`, at least for one of the two cases?